### PR TITLE
[FIX] account: convert amount in currency into debit/credit

### PR DESCRIPTION
**Pull request by jpp-odoo**
— Enabled multi currency;
— Create new journal entry;
— Add a new line;
— Add the column 'amount in currency';
— Select a different currency than the company currency;
— Set an amount in 'amount in currency'.

Before this commit, the amount in currency is not converted into debit
or credit fields.

Now, the amount is correctly converted into debit or credit fields.

opw-2129294

### CODE REVIEW

**Comment by nim-odoo** [comment]
runbot is failing, so I guess this is not ok :(

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -2879,12 +2887,23 @@ def _onchange_credit(self):
             self.debit = 0.0
         self._onchange_balance()
 
-    @api.onchange('amount_currency')
+    @api.onchange('amount_currency', 'currency_id')
```
Comment (smetl) [inline addons/account/models/account_move.py]: You need to manage that in `_onchange_currency`. You can't add this dependency there.

File: addons/account/models/account_move.py [hunk 2]
```diff
@@ -2879,12 +2887,23 @@ def _onchange_credit(self):
...
         for line in self:
             if not line.currency_id:
                 continue
             if not line.move_id.is_invoice(include_receipts=True):
+                # Recompute the debit/credit based on amount_currency/currency_id and date.
+                # However, date is a related field on account.move. Then, this onchange will not be triggered
+                # by the form view by changing the date on the account.move.
+                # To fix this problem, see _onchange_date method on account.move.
+
+                company_currency_id = line.account_id.company_id.currency_id
```
Comment (smetl) [inline addons/account/models/account_move.py]: Guidelines: Your variable must be `company_currency` if it's a record.

File: addons/account/models/account_move.py [hunk 3]
```diff
@@ -273,6 +273,14 @@ def _get_default_invoice_incoterm(self):
     # ONCHANGE METHODS
     # -------------------------------------------------------------------------
 
+    @api.onchange('date')
+    def _onchange_date(self):
```
Comment (smetl) [inline addons/account/models/account_move.py]: This is already managed by `_onchange_currency`, right?

**Comment by jpp-odoo** [comment]
@smetl Could you please check again ? Thanks

File: addons/account/models/account_move.py [hunk 4]
```diff
@@ -2905,7 +2906,19 @@ def _onchange_currency(self):
...
+                line._recompute_debit_credit_from_amount_currency()
 
+    def _recompute_debit_credit_from_amount_currency(self):
+        for line in self:
+            # Recompute the debit/credit based on amount_currency/currency_id and date.
+
+            company_currency = line.account_id.company_id.currency_id
+            amount = line.amount_currency
+            if line.currency_id and company_currency and line.currency_id != company_currency:
+                amount = line.currency_id._convert(amount, company_currency, line.account_id.company_id, line.date or fields.Date.today())
```
Comment (smetl) [inline addons/account/models/account_move.py]: line.move_id.date

Comment (smetl) [reply addons/account/models/account_move.py]: renamed your variable `balance` plz

**Comment by smetl** [comment]
@jpp-odoo Let me check functionaly plz

**Comment by smetl** [comment]
@robodoo r+

**Pull request closed without merging**